### PR TITLE
Replacing Cotter with Clerk

### PR DESCRIPTION
**Pull request by karllhughes**
Unfortunately, Cotter [was acquired and is shutting down](https://blog.cotter.app/blog/cotter-is-being-acquired-by-stytch), so I've got to replace the writer portal's login system.

I opted to use Clerk.dev because they're a client and their login system is similar to Cotter's.

@KSiig and @heraldofsolace could you two review this PR and test it out using the Replit invite I just sent to you?

Here's a video explaining how it should work (not much different from before): https://www.loom.com/share/d639d3a334834ed6b1b8086361ef5611

### CODE REVIEW

**Comment by heraldofsolace** [comment]
I don't seem to have received the Repl.it invite.

**Comment by KSiig** [comment]
I can't seem to find any invite either. I've logged in using the Google SSO but there's nothing there.

**Comment by karllhughes** [comment]
@KSiig @heraldofsolace hmm, I just emailed you an invite link, so let me know if that one works.

And when you get in, make sure you're testing this branch.

Thanks!

**Comment by heraldofsolace** [comment]
Received

**Comment by heraldofsolace** [comment]
I don't see the submit button.

Screenshot_20220106_181420

Other than that, it seems to work ok.

File: components/assignments/single-assignment.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+                console.error(e);
+                setUserData(false);
+            });
+    };
+
+    const belongsToCurrentUser = () => {
+        return currentUser && assignment.writer_email &&
+            assignment.writer_email.length > 0 &&
+            assignment.writer_email[0] &&
+            assignment.writer_email[0] === currentUser.identifier
```
Comment (heraldofsolace) [inline components/assignments/single-assignment.js:42]: `currentUser` doesn't seem to have `identifier` anymore

File: pages/api/writers/me.js [hunk 2]
```diff
@@ -1,42 +1,31 @@
...
-                           left join requests on requests.id = ANY (writers.requests)
-                           where writers.email like $1;`;
-            const {rows} = await pool.query(query, [decoded.payload.identifier]);
+      // Get user from Clerk API
+      const user = await users.getUser(req.session.userId);
+      // Get user
+      if (req.method === "GET") {
+          // Get assignment by assignmentId
+          const query = `select writers.id,
+                                writers.email,
```
Comment (heraldofsolace) [inline pages/api/writers/me.js:14]: We're returning `email` here, so it should be `currentUser.email` in `belongsToCurrentUser()`

**Comment by KSiig** [comment]
Apart from what @heraldofsolace already pointed out, it works fine for me.

**Comment by karllhughes** [comment]
@heraldofsolace good catch! And thank you both for reviewing.

I've updated this PR. I've still got some prep work to import the writer accounts, so let me know before this weekend if you have any other suggestions. Otherwise, I'll try to make this go live next week.

**Comment by karllhughes** [comment]
@KSiig @heraldofsolace this is now live!

Thanks for helping review it and let me know if you hear of any issues from writers.

**Pull request merged**